tests/integration/fns.sh

```diff
@@ -163,7 +163,22 @@ launch_evm() {
   fi
 }
 
-metallica
+ensure_process_count_equals() {
+  local process_name="$1"
+  local expected_count="$2"
+  local actual_count=$(ps aux | grep "$process_name" | grep -v "grep" | grep -v $$ | wc -l)
+  [ "$actual_count" -eq "$expected_count" ]
+  return $?
```

Comment:
This can still produce false positives. we should use `pgrep`, which was designed for this exact purpose. For instance:
> 
> ```bash
> local actual_count="$(pgrep -f "$process_name" | wc -l)"
> [ "$actual_count" -eq "$expected_count" ]
> return $?
> ```
> 
This avoids having to manually strip out `grep` processes and is less prone to accidental matches.

Reply:
Can also get false positives with pgrep as I actually was - which is switched to this as it is easier to debug!

Reply:
Alrighty!